internal/controller/controller.go

```diff
@@ -507,18 +509,68 @@ func (c *Controller) deployConfiguration(config *dynamic.Configuration) error {
 		return fmt.Errorf("unable to find any active mesh pods to deploy config : %+v", config)
 	}
 
-	for _, pod := range podList.Items {
-		log.Debugf("Deploying to pod %s with IP %s", pod.Name, pod.Status.PodIP)
+	if err := deployToPods(podList.Items, config); err != nil {
+		return fmt.Errorf("error deploying configuration: %v", err)
+	}
 
-		if deployErr := c.deployToPod(pod.Name, pod.Status.PodIP, config); deployErr != nil {
-			return fmt.Errorf("error deploying configuration: %v", deployErr)
+	return nil
+}
+
+// deployConfigurationToUnreadyNodes deploys the configuration to the mesh pods.
+func (c *Controller) deployConfigurationToUnreadyNodes(config *dynamic.Configuration) error {
+	podList, err := c.clients.ListPodWithOptions(c.meshNamespace, metav1.ListOptions{
+		LabelSelector: "component==maesh-mesh",
+	})
+	if err != nil {
+		return fmt.Errorf("unable to retrieve pod list: %v", err)
+	}
+
+	if len(podList.Items) == 0 {
+		return fmt.Errorf("unable to find any active mesh pods to deploy config : %+v", config)
+	}
+
+	var unreadyPods []corev1.Pod
+
+	for _, pod := range podList.Items {
+		for _, status := range pod.Status.ContainerStatuses {
+			if !status.Ready {
+				unreadyPods = append(unreadyPods, pod)
```

Comment:
If there are two unready containers in the pod, won't this result in the pod being appended twice?

Reply:
yeah, true.

Reply:
Nice catch, thanks !

Reply:
Actually no this code is correct. Once it hits the `break` statement it will break from the status loop and iterate to the next pod.

Reply:
True, I didnt see the break! nice